CMakeLists.txt

```diff
@@ -29,7 +29,7 @@ if(NOT CMAKE_BUILD_TYPE)
 endif()
 message(STATUS "CMAKE_BUILD_TYPE: ${CMAKE_BUILD_TYPE}")
 
-set(CMAKE_CXX_STANDARD 14 CACHE STRING "The C++ version to be used.")
+set(CMAKE_CXX_STANDARD 17 CACHE STRING "The C++ version to be used.")
```

Comment:
Could you follow
https://github.com/k2-fsa/k2/blob/2989b0b1186fa6022932804f5b39fbb2781ebf42/CMakeLists.txt#L181
to use
```
if(NOT CMAKE_CXX_STANDARD)
  set(CMAKE_CXX_STANDARD 14 CACHE STRING "The C++ version to be used.")
endif()
```

In setup.py from k2, we are doing
https://github.com/k2-fsa/k2/blob/2989b0b1186fa6022932804f5b39fbb2781ebf42/setup.py#L156
```python3
        major, minor = get_pytorch_version().split(".")[:2]
        major = int(major)
        minor = int(minor)
        if major > 2 or (major == 2 and minor >= 1):
            extra_cmake_args += f" -DCMAKE_CXX_STANDARD=17 "
```

Reply:
By the way, you can use
```
cmake -DCMAKE_CXX_STANDARD=17 ..
```
in the commandline with the above change if you are using torch 2.1+

Reply:
Done !

Reply:
@csukuangfj My bad, I did it but forgot to add it to the commit. I modify it accordingly